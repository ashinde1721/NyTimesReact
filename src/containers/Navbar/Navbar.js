import React, { Component } from 'react';
import './Navbar.scss';
import menuIcon from '../../assets/icons/white-menu.jpg';
import searchIcon from '../../assets/icons/search.png';
import closeIcon from '../../assets/icons/close.png';
import { connect } from 'react-redux';
import * as action from '../../actions/actions';
import { withRouter } from 'react-router-dom';

class Navbar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showSearch: true,
            showSearchInputBox: false
        }
    }

    searchInputHandler = (location) => {
        if (location.pathname !== '/') {
            this.setState({
                showSearch: false,
                showSearchInputBox: false
            })
        }
        else {
            this.setState({
                showSearch: true
            })
        }
    }

    openSearchTextBox = () => {
        this.setState({
            showSearchInputBox: true
        })
    }

    closeSearchTextBox = () => {
        this.props.onChange();
        this.setState({
            showSearch: true,
            showSearchInputBox: false
        })
    }

    componentDidUpdate(prevProps, prevState) {
        // only update chart if the location has changed
        if (prevProps.location !== this.props.location) {
            this.searchInputHandler(this.props.location);
            this.props.onChange();
        }
    }

    render() {
        let searchInputBox = (
            <span>
                <input type="text" className="display-inline form-control" placeholder="Search" onChange={this.props.onChange.bind(this)} />
                <img src={closeIcon} width="20" height="20" className="align" alt="" onClick={() => this.closeSearchTextBox()} />
            </span>
        );
        let defaultSearch = (
            <div>
                <img src={searchIcon} width="20" height="20" className="align" alt="" onClick={() => this.openSearchTextBox()} />
            </div>
        )
        return (
            <nav className="navbar aqua-marine" >
                <div className="navbar-brand white">
                    <div className="row">
                        <div className="col-6">
                            <img src={menuIcon} width="20" height="20" className="align" alt="" />
                            <span className={this.state.showSearchInputBox ? 'nav-title pad-left' : 'pad-left'}>NY Times Most Popular</span>
                        </div>
                        <div className="col-6">
                            <div className="float-right">
                                {
                                    (this.state.showSearch) ?
                                        (this.state.showSearchInputBox ? searchInputBox : defaultSearch)
                                        :
                                        null
                                }
                            </div>
                        </div>
                    </div>
                </div>
            </nav >
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        onChange: (e) => dispatch({
            type: action.SEARCH_TEXT,
            payload: {
                search: e ? e.target.value : null,
            }
        })
    }
}

export default connect(null, mapDispatchToProps)(withRouter(Navbar));