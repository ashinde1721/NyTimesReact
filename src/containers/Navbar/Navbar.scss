
.aqua-marine{
    background-color: #7fa6ff;
}

.white{
    color: white;
}

.align{
    margin-top: -4px;
}

.pad-left{
    padding-left: 15px; 
}

.navbar{
    position: fixed !important;
    z-index: 1;
}

.nav-title{
    font-size: 1.25rem;
    transition: 1s ease-in-out;
}

.navbar-brand{
    width: 100%;
}

.display-inline{
    display: inline !important;
}

@media (max-width: 576px) { 
    .nav-title{
        font-size: 0.8rem;
    }
 }