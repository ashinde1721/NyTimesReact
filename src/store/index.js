import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducer";
import createSagaMiddleware from 'redux-saga';

import {getArticleSaga} from  '../sagas/articlesSaga'

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(getArticleSaga);

export default store;