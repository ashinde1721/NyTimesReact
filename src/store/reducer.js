import * as actions from '../actions/actions';

const initialState = {
    search: '',
    articlesList: []
};

const reducer = (state = initialState, action) => {
    const { type, payload = {} } = action;

    switch (type) {
        case actions.SEARCH_TEXT:
            return { ...state, search: payload.search };

        case actions.SAVE_ARTICLES:
            return {
                ...state,
                articlesList: payload.articlesList
            }
        default:
            return state;
    }
};

export default reducer;
