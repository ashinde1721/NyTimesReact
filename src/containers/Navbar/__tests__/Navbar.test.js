import React from 'react';
import { Provider } from 'react-redux';

import renderWithRouter from '../../../config/renderWithRouter';
import store from '../../../store';
import Navbar from '../Navbar';

describe('Navbar Component', () => {
  it('renders component without breaking existing snapshot ui changes', () => {
    const { container } = renderWithRouter(
      <Provider store={store}>
        <Navbar />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
})