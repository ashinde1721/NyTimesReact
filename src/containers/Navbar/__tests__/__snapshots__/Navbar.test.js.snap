// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders component without breaking existing snapshot ui changes 1`] = `
<div>
  <nav
    class="navbar aqua-marine"
  >
    <div
      class="navbar-brand white"
    >
      <div
        class="row"
      >
        <div
          class="col-6"
        >
          <img
            alt=""
            class="align"
            height="20"
            src="white-menu.jpg"
            width="20"
          />
          <span
            class="pad-left"
          >
            NY Times Most Popular
          </span>
        </div>
        <div
          class="col-6"
        >
          <div
            class="float-right"
          >
            <div>
              <img
                alt=""
                class="align"
                height="20"
                src="search.png"
                width="20"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
